ui/register_react_component.md

```diff
@@ -0,0 +1,98 @@
+When you want to create new instance of recently created component just call newInstance function with these arguments
+* name - name of component you wish to use.
+* props - object with props which will be mapped to your components (used when creating blueprint)
+* element - DOM element where this new instance of component should be rendered.
+```JS
+import newInstance from 'registry';
+
+const props = {};
+const element = document.getElementById('element-id');
+newInstance('YourComponentName', props, element);
+```
+
+### Shared component API
+Abstract shared components has API controlling active components, this way you can define and create new component, later you can access to all active components on screen and use it's shared interface to interact with it.
+
+#### Define
+When you want to define new component to be used by someone call define function with name of such component and it's blueprint.
+```JS
+import define from 'registry';
+
+define('YourComponentName', componentBlueprint);
+```
+
+#### Get definition
+To fetch definition of component you created you can call specific function. You will get name of such component and it's blueprint. With such blueprint you can call functions to update, create or remove it.
+```JS
+import getDefinition from 'registry';
+
+getDefinition('YourComponentName');
+```
+
+#### New instance
+To create new instance and mount it to DOM element call function `newInstance`. This function not only creates new instance it also gives you object over which you can update it's pros, update it or destroy it from component's drawer. Inside this object you will also find comeponent's instance ID and should the component chooses it's interact property for side channel communication.
+
+When creating new instance you have option to pass multiple parameters:
+* name - component's name under which component was defined
+* props - default props to component
+* element - DOM element under which this component should be rendered (optional)
+```JS
+import newInstance from 'registry';
+
+newInstance('YourComponentName', {}, element);
+```
+
+#### Get instance
+To receive component's instance from registery you can call function `getInstance`. You will get full information about this instance, ways to interact with it and it's name and ID.
+```JS
+import getInstance from 'registry';
+
+getInstance('YourComponentName', componentID);
+```
+
+#### Is defined
+Check if component with some name is defined in drawer. Great for checking if component's name is available.
+```JS
+import isDefined from 'registry';
+
+isDefined('YourComponentName');
+```
+
+### Component instance API
+When you create new instnace or search for instance of component you will receive object over which you can interact with this instance and observe it's properties.
+
+#### ID
+Instance's ID under which it's created.
+
+#### interact
+Should the component chooses it can supply object over which it's possible side channel communication.
```

Comment:
s/chooses/choose/

which/that?

"possible side channel communication" -- verb missing?